babeld/babel_interface.c

```diff
@@ -353,51 +360,51 @@ DEFUN (babel_split_horizon,
     babel_ifp = babel_get_if_nfo(ifp);
 
     assert (babel_ifp != NULL);
-    babel_ifp->flags |= BABEL_IF_SPLIT_HORIZON;
+    if (no)
+        babel_ifp->flags &= ~BABEL_IF_SPLIT_HORIZON;
+    else
+        babel_ifp->flags |= BABEL_IF_SPLIT_HORIZON;
     return CMD_SUCCESS;
 }
 
-/* [Interface Command] Disable split horizon (default). */
-DEFUN (no_babel_split_horizon,
-       no_babel_split_horizon_cmd,
-       "no babel split-horizon",
-       NO_STR
+/* [Interface Command]. */
+DEFPY (babel_set_hello_interval,
+       babel_set_hello_interval_cmd,
+       "babel hello-interval (20-655340)",
        "Babel interface commands\n"
-       "Disable split horizon processing\n")
+       "Time between scheduled hellos\n"
+       "Milliseconds\n")
 {
     VTY_DECLVAR_CONTEXT(interface, ifp);
     babel_interface_nfo *babel_ifp;
 
     babel_ifp = babel_get_if_nfo(ifp);
-
     assert (babel_ifp != NULL);
-    babel_ifp->flags &= ~BABEL_IF_SPLIT_HORIZON;
+
+    babel_ifp->hello_interval = hello_interval;
     return CMD_SUCCESS;
 }
 
-/* [Interface Command]. */
-DEFUN (babel_set_hello_interval,
-       babel_set_hello_interval_cmd,
-       "babel hello-interval (20-655340)",
+DEFPY (no_babel_set_hello_interval,
+       no_babel_set_hello_interval_cmd,
```

Comment:
Can't this be combined with `babel_set_hello_interval_cmd`?

Reply:
Sure, I've combined all of those similar commands.